Var/src/var.c

```diff
@@ -0,0 +1,84 @@
+#include "var.h"
+#define PRINT_VAR(var_type, format) printf(""#format" (%p)\n", *(var_type*)ptr->valptr, ptr->valptr)
+
+#define OPTIMIZED_REALLOC(var_type, old_size, size) \
```

Comment:
`realloc` handles the case where `size > old_size` and `free`s and then `malloc`s: `man 3 realloc`

Reply:
Great, ill read about that. What I did was based on
stackoverflow.com/questions/1401234/differences-between-using-realloc-vs-free-malloc-functions answer number 2

Reply:
I think you misunderstood the answer. Read this answer: https://stackoverflow.com/a/1401247/10337548
 
Using `free()` and then `malloc()` to change the size of the array is *very bad design* as no one promised you that `malloc()` will return the same pointer (meaning you get a completely new array with different data). When you allocate a new array with `malloc()`, it gives you a (probably) new pointer to an array of that size (or NULL but that's not for now), while `realloc()` is used to change the size of the array since it will also copy the old contents if the array was moved.